rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,233 @@
+### DCP PR:
```

Comment:
The main questions that I have upon reading this are high level: 

1. Why does our existing bundle structure not work? I want to be convinced that the bundles for Pe'er and ICA weren't created incorrectly, or created based on faulty metadata or data file headers.
2. Second, why can an "update" in the context of the ongoing AUDR work **not** fix this issue by generating a single bundle containing all files to be processed together?

Reply:
On the second point, at the very least there's currently no way to indicate bundle lineage except by versioning in the datastore. This only allows a single parent for a child.

Reply:
Sorry to re-open this "resolved" conversation. @samanehsan and I have talked about Ambrose's first comment above, i.e. 
>  I want to be convinced that the bundles for Pe'er and ICA weren't created incorrectly, or created based on faulty metadata or data file headers.

Based on my own understanding, we are and will be facing at least 3 problems: 
a. pipelines need to process data span multiple data bundles
b. pipelines need to process data span multiple projects
c. data submitters (e.g. labs) want to submit supplementary data in addition to their original, completely submitted data after a certain time period. 

The case `c.` is tricky and I believe most of the data platforms will inevitably face a similar issue. I also believe this RFC proposed a good and feasible way to deal with that scenario.

The case `b.` to me is a perfect candidate to be resolved by this RFC, since we don't have a mechanism to enable pipelines process across projects, unless the submitters making a weird submission with data from multiple projects intentionally.

When it comes to case `a.`, this RFC pointed out that
> The current DCP model of processing has a scope of one data bundle, which contain a single assay or analysis. There is no mechanism to know when a group of bundles that need to be processed as a group incomplete and available.

I'm not super familiar with how the bundles are created from the spreadsheet at the upstream service, but the pipelines have to process multiple bundles in a single project together, does it mean the algorithm/rule that deciding which files(fastqs + metadata) go to which bundle needs to be updated at the first place (prior to DSS), instead of adding extra layers to group the created bundles in downstream services? This is my confusion, it'd be great if someone knows more about the bundle creation mechanism can point out what I missed there if I'm wrong!

Reply:
@rexwangcc thanks for reopening.

I will let @justincc explain the bundling algorithm.  

In my view, bundles are a storage concept, not an experimental concept.   No matter what the bundling policy is, per-bundle processing give ingest control of how analysis experiments are designed.  Analysis should have the ability to construct experiments in any way they see fit without the need to change the storage model.

Reply:
Hey @rexwangcc. Here's [a doc on bundling](https://docs.google.com/document/d/1cGfolHMGEe1IKkXSBS75SAdSri9Np_oW6Z4S9-Ax1QY).